checkov/github_actions/utils.py

```diff
@@ -75,18 +75,22 @@ def is_schema_valid(config: dict[str, Any] | list[dict[str, Any]]) -> bool:
 
 
 def get_gha_files_definitions(root_folder: str | Path,
-                              files: "list[Path] | None" = None,
+                              files: "list[str] | None" = None,
                               runner_filter: RunnerFilter | None = None,) -> tuple[dict[str, Any], dict[str, Any]]:
     definitions = {}
     definitions_raw = {}
     file_paths = get_scannable_file_paths(root_folder=root_folder)
+    files_set = set(files) if files else set()
 
     for file_path in file_paths:
-        result = parse_file(f=file_path)
-        # result should be tuple of dict representing the file payload structure and list of lines of the payload
-        if result is not None:
-            definitions[str(file_path)] = result[0]
-            definitions_raw[str(file_path)] = result[1]
+        str_file_path = str(file_path)
+        process = str_file_path in files_set if files_set else True
+        if process:
```

Comment:
@maxamel Please use a prefix that hints this is a boolean. Also, `process` is about as generic as it should be. Perhaps `should_parse` is a better name?